### PR TITLE
[WIP] AD libMesh

### PR DESCRIPTION
**Pull request by lindsayad**
Refs #2121

### CODE REVIEW

**Comment by lindsayad** [comment]
This is essentially the non-templated less-disruptive version of #2396

**Comment by jwpeterson** [comment]
This PR has been marked "do not merge" since we are no longer accepting PRs into the `master` branch. All new PRs should be made on the `devel` branch instead. Once this PR's target branch has been updated to `devel`, the "do not merge" label will be removed.

File: include/fe/fe_base.h [hunk 1]
```diff
@@ -116,14 +117,22 @@ class FEGenericBase : public FEAbstract
...
    */
-  typedef OutputType                                                      OutputShape;
-  typedef typename TensorTools::IncrementRank<OutputShape>::type          OutputGradient;
-  typedef typename TensorTools::IncrementRank<OutputGradient>::type       OutputTensor;
-  typedef typename TensorTools::DecrementRank<OutputShape>::type          OutputDivergence;
-  typedef typename TensorTools::MakeNumber<OutputShape>::type             OutputNumber;
-  typedef typename TensorTools::IncrementRank<OutputNumber>::type         OutputNumberGradient;
-  typedef typename TensorTools::IncrementRank<OutputNumberGradient>::type OutputNumberTensor;
-  typedef typename TensorTools::DecrementRank<OutputNumber>::type         OutputNumberDivergence;
+  typedef typename MakeOutput<OutputType>::type                             OutputShape;
```
Comment (lindsayad) [inline include/fe/fe_base.h:120]: I think that `OutputShape` should be exactly the same as `OutputType`

Comment (lindsayad) [reply include/fe/fe_base.h:120]: But maybe not ... shape evaluations are a function of chi, eta, etc.... and those *could* be a function of mesh displacement

**Comment by lindsayad** [comment]
I don't see a funding path for continuing this and we've gone beyond the 4 year mark since opening. Closing this for my bookkeeping

**Pull request closed without merging**